search_knn.go

```diff
@@ -293,18 +289,21 @@ func mergeKNN(req *SearchRequest, sr *SearchResult) {
 	sr.Total = uint64(len(nonZeroScoreHits))
 }
 
-func recomputeTotalScore(operator int, hit *search.DocumentMatch) float64 {
+func recomputeTotalScore(operator knnOperator, hit *search.DocumentMatch) float64 {
 	totalScore := 0.0
-	numNonZero := 0
+	numNonZero := 0.0
+	numTotal := float64(len(hit.ScoreBreakdown))
 	for _, score := range hit.ScoreBreakdown {
 		if score != 0 {
 			numNonZero += 1
 		}
 		totalScore += score
 	}
-	if operator == 0 {
-		coord := float64(numNonZero) / float64(len(hit.ScoreBreakdown))
+	if operator == knnOperatorOr || operator == "" {
+		coord := numNonZero / numTotal
```

Comment:
Can't numTotal be zero in case the request carried "score":"none" ?

Reply:
Suppose this is applicable only to knn scoring, so scoreBreakdown is expected to have at least one score even if score:none. This area will need revisiting later.